From 77c67d0e597f5408c3d005ef624ef3d417271e0e Mon Sep 17 00:00:00 2001
From: sunguoshuai <sunguoshuai@huawei.com>
Date: Thu, 9 Sep 2021 21:34:40 +0800
Subject: [PATCH] fix not striped

---
 appliance/Makefile.am | 1 +
 1 files changed, 1 insertions(+)

diff -ruN libguestfs/appliance/Makefile.am libguestfs-1.49.5/appliance/Makefile.am
--- libguestfs/appliance/Makefile.am	2022-11-16 02:11:11.123199053 +0800
+++ libguestfs-1.49.5/appliance/Makefile.am	2022-11-16 02:13:36.113261902 +0800
@@ -118,6 +118,7 @@
 	rm -rf tmp-d
 	mkdir -p tmp-d$(DAEMON_SUPERMIN_DIR) tmp-d/etc
 	ln ../daemon/guestfsd tmp-d$(DAEMON_SUPERMIN_DIR)/guestfsd
+	strip tmp-d/usr/sbin/guestfsd
 	( cd tmp-d && tar zcf - * ) > $@-t
 	rm -r tmp-d
 	mv $@-t $@
